scripts/motion_planner.py

```diff
@@ -9,47 +13,92 @@ class MotionPlanner:
     def __init__(self, _action_server):
         self._action_server = _action_server
         self.task = None
+        self.velocity_pub = rospy.Publisher('movement_velocity_targets', TwistStampedArrayStamped, queue_size=0)
+        self.arm_service = rospy.ServiceProxy('uav_arm', SetBool)
 
-    def get_velocity_command(self):
+    def run(self):
+        rate = rospy.Rate(10)
+        while not rospy.is_shutdown():
+            # Tuples could have different lengths so just get the whole tuple
+            task_command = self._get_task_command()
+            if(task_command[0] == 'velocity'):
+                self._publish_twist(task_command[1])
+            elif(task_command[0] == 'arm'):
+                task_command[1](self._use_arm_service(True))
+            elif(task_command[0] == 'disarm'):
+                task_command[1](self._use_arm_service(False))
+            elif(task_command[0] == 'nop'):
+                self._publish_twist(TwistStamped())
+            else:
+                rospy.logerr('Unkown command request: %s, noping', task_command[0])
+                self._publish_twist(TwistStamped())
+            rate.sleep()
+
+    def _use_arm_service(self, arm):
+        try:
+            armed = self.arm_service(arm)
+        except rospy.ServiceException as exc:
+            rospy.logerr("Could not arm: " + str(exc))
+            armed = False
+        return armed
+
+    def _publish_twist(self, twist):
+        velocity_msg = TwistStampedArrayStamped()
+        velocity_msg.header.stamp = rospy.Time.now()
+        velocity_msg.data = [twist]
+        self.velocity_pub.publish(velocity_msg)
+
+    def _get_task_command(self):
         if (self.task is None) and self._action_server.has_new_task():
             self.task = self._action_server.get_new_task()
 
         if self.task:
             if self._action_server.is_canceled():
                 self.task.cancel()
```

Comment:
Exceptions from the action server? Im ok with letting the motion planner crash at that point.